### PR TITLE
Basic solutions

### LINKED ISSUE #46: A solution for basic04 would be helpful
First, thank you for these tutorial resources. It's awesome that you're putting this out there.

I am struggling with basic04 assignment 1. After a certain amount of beating my head against the wall, it would be helpful if I could look at a solution (or an example program which solves this problem) to understand the changes and move on. However, the tutorial doesn't have a solution for this step. After a fair amount of digging around, I thought about skipping this assignment and moving on, but the next step in the tutorial (packet01) pulls in *our* solution from basic04 via Makefile, so it seems like I cannot move on without solving it. 

So I am not asking for things empty-handed, here are the things that I have tried:
1. Use `bpf_obj_get` to retrieve the file descriptor for the pinned map by supplying the pathname to the pinned map file in bpffs. I do not understand why, but each call to `bpf_obj_get` increments the fd by 1. So if my initial map_fd is 3, `bpf_obj_get` will return 4, 5, 6, etc... Some [other documentation](https://prototype-kernel.readthedocs.io/en/latest/bpf/ebpf_maps.html) recommended [this program](https://github.com/netoptimizer/prototype-kernel/blob/master/kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c) for example usage of `bpf_obj_get` and has a very similar format to our program, but it also does not solve this specific problem (it is a TODO).
2. The assignment suggests to remember the original map_id. So I took the map_id from the info struct populated by the original call to `check_map_fd_info` in main, and passed it into the `stats_poll` function. For each poll loop, I use `bpf_obj_get_info_by_fd` to see if the map_id has changed at that particular file descriptor. However, it never changes, despite me loading a new BPF map with xdp_loader. I assume the problem is I am getting the map_id from the original map file descriptor, and the new map has a different file descriptor. If this is true, that would imply I need to probe when the file descriptor changes, which takes me back to the solution I tried previously and does not work.

Although having a solution in the tutorial would probably be best to ensure the tutorial is self contained, that may be more appropriate in the long term. In the short term, any help is appreciated. Thanks.

### PR DESCRIPTION
**Pull request by aspsk**
Add solutions for basic04/* assignments and fix some simple issues. This PR fixes #46.

### CODE REVIEW

File: common/common_user_bpf_xdp.c [hunk 1]
```diff
@@ -244,3 +230,40 @@ int check_map_fd_info(int map_fd, struct bpf_map_info *info,
...
+	int err, len, fd;
+	__u32 info_len = sizeof(*info);
+
+	len = snprintf(filename, PATH_MAX, "%s/%s", pin_dir, mapname);
+	if (len < 0) {
+		fprintf(stderr, "ERR: constructing full mapname path\n");
+		return -1;
+	}
+
+	/* Lesson#1: There is only a weak dependency to libbpf here as
```
Comment (tohojo) [inline common/common_user_bpf_xdp.c]: Not sure if this comment makes sense anymore. Certainly the lession#1 part should go :)

Comment (aspsk) [reply common/common_user_bpf_xdp.c]: @tohojo I removed this comment, as other functions in the file do depend on `libbpf`

Comment (tohojo) [reply common/common_user_bpf_xdp.c]: OK, SGTM. I don't see that in the PR, though, did you push your updated branch?

Comment (aspsk) [reply common/common_user_bpf_xdp.c]: @tohojo thank, just pushed

**Pull request merged**